src/hooks/__tests__/use-track-component.test.tsx

```diff
@@ -0,0 +1,23 @@
+import {renderHook, act} from '@testing-library/react-hooks'
+import {useTrackComponent} from '../use-track-component'
+import {track} from 'utils/analytics'
+
+jest.mock('utils/analytics')
+
+beforeEach(() => {
+  jest.clearAllMocks()
+})
```

Comment:
This uses Jest's automatic mocking to enable the tests to see if `track` gets called.

https://jestjs.io/docs/es6-class-mocks#automatic-mock